infer/environment.cc

```diff
@@ -203,6 +215,18 @@ KnowledgeFact &KnowledgeRef::mutate() {
     return *knowledge.get();
 }
 
+void KnowledgeRef::removeReferencesToVar(cfg::LocalRef var) {
+    if (typeTestReferencesVar((*this)->yesTypeTests, var) || typeTestReferencesVar((*this)->noTypeTests, var)) {
```

Comment:
Why `(*this)->yesTypeTests` instead of `knowledge->yesTypeTests`? There's no special behavior inside of `KnowledgeRef::operator->`, so it should be reasonable to use the field directly instead.

```suggestion
    if (typeTestReferencesVar(knowledge->yesTypeTests, var) || typeTestReferencesVar(knowledge->noTypeTests, var)) {
```

Reply:
Done!